### ISSUE TITLE
convoy-nfs instances keeps restarting after host went from "inactive" to "active" state.

### ISSUE THREAD

**Issue by sangeethah** [rancher/rancher#3154]
Server version - V0.51.0

Had a  convoy-nfs stack deployed from catalog.
Add a new host.

Make sure that the convoy-nfs instance gets started successfully on this instance.
Now deactivate this host.

We see the convoy-nfs instances getting removed since they get marked unhealthy.

Now activate this host.

When host is activated , we see a new convoy-nfs instance that gets deployed. But this instance keeps restarting and following exception is seen in logs:

```
12/30/2015 4:20:00 PMWaiting for metadata.time="2015-12-31T00:20:00Z" level=info msg="Execing [/usr/bin/nsenter --mount=/proc/3691/ns/mnt -F -- /var/lib/docker/aufs/mnt/b4d848c668978a54dc9c143764c7702f0bcebfc0c491a9425a36a5e35b6df435/var/lib/rancher/convoy-agent/share-mnt --stage2 /var/lib/rancher/convoy/convoy-nfs-3dc6f1fc-d61b-4607-813b-60a762becfa7 -- /launch volume-agent-nfs-internal]"
12/30/2015 4:20:00 PMRegistering convoy socket at /var/run/conoy-convoy-nfs.sock
12/30/2015 4:20:00 PMMounting at: /var/lib/rancher/convoy/convoy-nfs-3dc6f1fc-d61b-4607-813b-60a762becfa7/mnt
12/30/2015 4:20:00 PMmkdir: cannot create directory '/var/lib/rancher/convoy/convoy-nfs-3dc6f1fc-d61b-4607-813b-60a762becfa7/mnt': File exists
```

**Comment by fabn** [comment]
Getting the same issue, any workaround or quick fix?

**Comment by deniseschannon** [comment]
Are you now running v0.56.1 with the new convoy-nfs version (v0.2)? I haven't had time to try out this set up to reproduce.

**Comment by marvinwankersteen** [comment]
Getting exactly the same issue with v0.59.1 and convoy-nfs v0.2.

`1.3.2016 13:52:00Waiting for metadata.
1.3.2016 13:52:00time="2016-03-01T12:52:00Z" level=info msg="Execing [/usr/bin/nsenter --mount=/proc/7150/ns/mnt -F -- /var/lib/docker/aufs/mnt/a8f202c4980ef423e5dd5ff5b7e11480c8358efee1a039e8e354704fd8c60cc8/var/lib/rancher/convoy-agent/share-mnt --stage2 /var/lib/rancher/convoy/NFS-bbfb6124-3bac-4672-a943-190798ebe0d7 -- /launch volume-agent-nfs-internal]"
1.3.2016 13:52:03Registering convoy socket at /var/run/convoy-NFS.sock
1.3.2016 13:52:03Mounting at: /var/lib/rancher/convoy/NFS-bbfb6124-3bac-4672-a943-190798ebe0d7/mnt
1.3.2016 13:52:03mkdir: cannot create directory '/var/lib/rancher/convoy/NFS-bbfb6124-3bac-4672-a943-190798ebe0d7/mnt': File exists`

**Comment by withinboredom** [comment]
For what its worth:

On the host (/var/lib/rancher/convoy/convoy-nfs-1a5e52be-5d03-4b4b-bd31-172fc561be70) I see mnt cannot be deleted ... Input/Output Error

It takes a reboot to be able to delete it.

**Comment by marvinwankersteen** [comment]
You can just umount the mnt-directory. I had the same issue and I could delete it without rebooting.

**Comment by withinboredom** [comment]
I guess it depends if nfs happened to close the descriptor or not.

**Comment by cjellick** [comment]
`mount -l -f /path` seems to work and convoy-nfs recovers. But (at least on RancherOS), this leads to CPU spike (and staying at) close to 100% with `[kworker]` being the consumer.

**Comment by cjellick** [comment]
So, I'm thinking that I can fix this by `umount`ing before `mount`ing in the convoy-nfs startup script, but I need to get to the bottom of the CPU spike -it seems to actually really degrade performance.

**Comment by cjellick** [comment]
After upgrading to the latest version of convoy (`v0.4.4-rc2`) and making a number of other changes to convoy-agent, I cannot reproduce this. So, I'm calling it fixed.

**Comment by sangeethah** [comment]
Tested with latest build from master and convoy agent - 

Deactivating the host , does not result in the nfs-convoy agent to be marked "unhealthy"  and destroyed as stated in the bug. Must be an issue that was resolved in the later releases.

I have tried to power off and power on the host where nfs-convoy agent  is running , so that the nfs-convoy agent gets deleted and recreated in the same host.

convoy-nfs instance is able to start successfully with out any issues after the host is powered on.
